packages/blitz-auth/src/server/adapters/next-auth/internals/core/lib/cookie.ts

```diff
@@ -0,0 +1,236 @@
+import type {CookiesOptions} from "next-auth"
```

Comment:
we need **all** of these files that are next-auth/internals?

Reply:
From my experimentation with the next auth patch. We need to import only the 3 function the oAuthCallback for the callback the getAuthorizedUrl for the login and the init to generate the request and the options to the required way next auth requires.

These other files I have copied are all dependencies of those functions.